Add files via upload

Gediminas-msc opened the pull request:
For the assignment "Export GitHub project as csv table“.

Gediminas-msc committed:
Add files via upload

sirex commented on scripts/Github_project_to_CSV.py:
```diff
@@ -0,0 +1,124 @@
+from pprintpp import pprint as pp
+
+# https://github.com/settings/tokens/819269469
+token = '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'
```
Remove token and move it as an command argument. In order to do that, you need to move everything under `main` function, you can see how this can be done by using this example:

https://github.com/atviriduomenys/katalogas/blob/devel/scripts/anonymize.py

Gediminas-msc committed:
Update Github_project_to_CSV.py

Changes associated with Typer usage in order to use token as an argument.

sirex committed:
Rename github export script

sirex commented on scripts/github-export.py:
```diff
@@ -0,0 +1,132 @@
+"""
+poetry init -n
+poetry add requests
```
This should be moved to `pyproject.tom` as dev dependency. And whole this comment should be removed.

Gediminas-msc replied:
Comments removed.

sirex commented on scripts/github-export.py:
```diff
@@ -0,0 +1,132 @@
+        }
+    }
+    '''}).json())
+    # | {'id': 'PN_kwDOAKaBNM4AGx3p', 'title': 'Palaikymas'},
```
Please remove debug comments. And add another argument - `project`, if `project` is not given then list all projects with id ant title, and exit.

If project id is given, then do not list projects and proceed with the export.

Gediminas-msc replied:
Comments removed. New argument `project_id` added.

sirex commented on scripts/github-export.py:
```diff
@@ -0,0 +1,132 @@
+    '''}).json())
+    # | {'id': 'PN_kwDOAKaBNM4AGx3p', 'title': 'Palaikymas'},
+
+    with open('/tmp/Github_project_to_csv_file.csv', 'w') as f_csv:
```
CSV file should be printed to stdout.

Gediminas-msc replied:
`writer = csv.writer(sys.stdout)`

Gediminas-msc committed:
Update github-export.py

sirex commented on scripts/github-export.py:
```diff
@@ -0,0 +1,130 @@
+from typer import Argument
+from typer import run
+
+import requests
```
This package should be added to `pyproject.toml`:

    poetry add -G dev requests

sirex commented on scripts/github-export.py:
```diff
@@ -0,0 +1,130 @@
+
+def main(
+        token: str = Argument(..., help='https://github.com/settings/tokens'),
+        project_id: Optional[str] = Argument(None)
```
Please, add `help` message, what it is and how to use it.

Gediminas-msc committed:
Update github-export.py

Gediminas-msc committed:
Add requests dependency.

sirex reviewed (changes requested):
```python
$ poetry run python scripts/github-export.py --help
Traceback (most recent call last):
  File "scripts/github-export.py", line 11, in <module>
    from pprintpp import pprint as pp
ModuleNotFoundError: No module named 'pprintpp'
```

Gediminas-msc committed:
Add pprintpp dependency and minor changes.

sirex committed:
Finish github export script

Now this script calculates all the needed numbers and prints it into a
nicely formatted table.

sirex committed:
Fix poetry.lock

sirex committed:
Fix export to csv
